projects/components/src/toggle-switch/toggle-switch.component.ts

```diff
@@ -9,6 +10,7 @@ import { MatSlideToggleChange } from '@angular/material/slide-toggle';
     <mat-slide-toggle
       color="primary"
       [(ngModel)]="this.checked"
+      [ngClass]="{ 'small-slide-toogle': this.size === ${ToggleSwitchSize.Small} }"
```

Comment:
Is it feasible to apply a size class either way (`small-slide-toggle` or `medium-slide-toggle`? That way, we can update the scss such that small styles don't just apply as overrides to the medium styles. The overriding behavior makes the css harder to understand and more error prone, in my opion.

Reply:
By default, the slide-toggle has a medium size, for that reason, I'm overriding the styles. Tell me a bit more about your approach, I think that I don't understand your point.

Reply:
> By default, the slide-toggle has a medium size

So in my mind, that knowledge of the default should not extend beyond the line `public size?: string = 'small';`. If we were to change that to default to medium, that shouldn't require css changes (currently it wouldn't _require_ them, but the css wouldn't make much sense having the override behavior it does).

Regardless of the default though, the reason is just to have symmetry in the implementation for maintainability. A code-level equivalent of this (with the same issue) would be to say create a `Shape` class, which returns an area of `length * width`, a `Rectangle` subclass which doesn't override area, and a `Circle` subclass which does (to `pi*r^2`). Similarly, there it would make sense to make area abstract and have each shape implement it itself.